dotCMS/src/main/java/com/dotcms/storage/FileStorageAPIImpl.java

```diff
@@ -0,0 +1,338 @@
+
+        return this.generateFullMetaData(binary, s -> true, maxLength); // raw = no filter
+    }
+
+    /**
+     * {@inheritDoc}
+     * Stand alone Tika-independent metadata
+     * We can get it without having to call tika
+     * @param binary {@link File} file to get the information
+     * @param metaDataKeyFilter  {@link Predicate} filter the meta data key for the map result generation
+     * @return
+     */
+    @Override
+    public Map<String, Serializable> generateBasicMetaData(final File binary,
+            final Predicate<String> metaDataKeyFilter) {
+
+        final ImmutableSortedMap.Builder<String, Serializable> mapBuilder =
+                new ImmutableSortedMap.Builder<>(Comparator.naturalOrder());
+
+        if (this.validBinary(binary)) {
+
+            mapBuilder.put(TITLE_META_KEY, binary.getName());
+            final String relPath = binary.getAbsolutePath()
+                    .replace(ConfigUtils.getAbsoluteAssetsRootPath(),
+                            StringPool.BLANK);
+
+            mapBuilder.put(PATH_META_KEY, relPath);
+
+            if (metaDataKeyFilter.test(LENGTH_META_KEY)) {
+                mapBuilder.put(LENGTH_META_KEY, binary.length());
+            }
+
+            if (metaDataKeyFilter.test(CONTENT_TYPE_META_KEY)) {
+                mapBuilder.put(CONTENT_TYPE_META_KEY, MimeTypeUtils.getMimeType(binary));
+            }
+
+            mapBuilder.put(MOD_DATE_META_KEY, binary.lastModified());
+            mapBuilder.put(SHA226_META_KEY,
+                    Try.of(() -> FileUtil.sha256toUnixHash(binary)).getOrElse("unknown"));
+
+        }
+
+        return mapBuilder.build();
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param binary  {@link File} file to get the information
+     * @param metaDataKeyFilter  {@link Predicate} filter for the map result generation
+     * @param maxLength {@link Long} max length is used when parse the content, how many bytes do you want to parse.
+     * @return
+     */
+    @Override
+    public Map<String, Serializable> generateFullMetaData(final File binary,
+            final Predicate<String> metaDataKeyFilter,
+            final long maxLength) {
+
+        final TreeMap<String, Serializable> metadataMap = new TreeMap<>(Comparator.naturalOrder());
+
+        try {
+            final Map<String, Serializable> fullMetaDataMap = this.metadataGenerator.generate(binary, maxLength);
+            if (UtilMethods.isSet(fullMetaDataMap)) {
+                for (final Map.Entry<String, Serializable> entry : fullMetaDataMap.entrySet()) {
+                    if (metaDataKeyFilter.test(entry.getKey())) {
+                        metadataMap.put(entry.getKey(), entry.getValue());
+                    }
+                }
+            }
+            //basic meta data should override any previous value that might have exist already.
+            metadataMap.putAll(this.generateBasicMetaData(binary, metaDataKeyFilter));
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)